### DISCUSSION TITLE
Support several Accounts of the same CoinType in StoredKey

### DISCUSSION THREAD

**Discussion by grachyov** [trustwallet/wallet-core#1572] [Q&A]
I would like to have several accounts of the same CoinType derived from mnemonic.

Would it be ok to extend `StoredKey` and `Wallet` functions taking `CoinType` as a parameter with a new parameter: account's index, while keeping current functions' signatures for backward compatibility?

**Comment by optout21** [comment]
There is a method on HDWallet to derive a key with custom derivation path, can this be used for your use case?

```
struct TWPrivateKey *_Nonnull TWHDWalletGetKey(struct TWHDWallet *_Nonnull wallet, enum TWCoinType coin, TWString *_Nonnull derivationPath);
```

**Reply by grachyov** [reply]
This way I won’t be able to get these accounts later from StoredKey.

**Reply by optout21** [reply]
That's true.  However, in Trust Wallet / Wallet Core we have no plans to support multiple accounts (of the same coin) in a wallet, so if this is a feature request, it is rejected.

**Reply by grachyov** [reply]
Could we discuss how can I add multiple accounts support into WalletCore for wallets needing this feature?

**Reply by vikmeup** [reply]
> That's true. However, in Trust Wallet / Wallet Core we have no plans to support multiple accounts (of the same coin) in a wallet, so if this is a feature request, it is rejected.

@catenocrypt 

Is this backward compatible feature? 
What is the complexity for having multiple accounts in one wallet
